File: server/python/source/message.py
class Message:
    """
    Message class represents a message with its metadata.

    Attributes:
        correlationId (str): The unique identifier for the message.
        message (str): The content of the message.
        from_field (str): The sender of the message.
        to_field (str): The recipient of the message.
        timestamp (float): The time the message was created.

    Methods:
        __str__(): Returns a string representation of the message.
    """

    def __init__(self, correlationId, message, from_field, to_field, timestamp):
        self.correlationId = correlationId
        self.message = message
        self.from_field = from_field
        self.to_field = to_field
        self.timestamp = timestamp

    def __str__(self) -> str:
        return (
            f"\nmessage: {self.message}\nfrom: {self.from_field}\nto: {self.to_field}"
        )

File: server/dotnet/server/src/Message.cs
namespace server.src
{
    /// <summary>
    /// Represents a message with a correlation ID, content, sender, recipient, and timestamp.
    /// </summary>
    /// <param name="correlationId">The correlation ID of the message.</param>
    /// <param name="content">The content of the message.</param>
    /// <param name="from">The sender of the message.</param>
    /// <param name="to">The recipient of the message.</param>
    /// <param name="time">The timestamp of the message.</param>
    internal class Message(uint correlationId, string content, string from, string to, ulong time)
    {
        public uint CorrelationId { get; } = correlationId;
        public string Content { get; } = content;
        public string From { get; } = from;
        public string To { get; } = to;
        public ulong Time { get; } = time;
    }
}
